### ISSUE TITLE
Projectroles: Rework project creation and ownership permissions under categories

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-core#464]
..or move them from another category. Depends on #463.

The moving is, from the technical side, a completely separate functionality and will be handled in a separate ticket.

This one actually creates an issue: if we freely let users create projects under a category with themselves as the owner, this can create a situation where the owner may not have access to projects below a category they own. I see two ways to get around this:

- Owner/delegate forcing
    * Force the owner of freshly created projects into the owner of the parent category, with the creating user set up automatically as a delegate
- Ownership inheritance
    * Implement an inheritance structure where an owner of a category has owner access to all sub-categories and projects beneath
    * Needs to be communicated to the user in the member UI
    * Related to #17

The latter might be a bit more work, but it would also be logical, consistent and easily communicateable to the user. For now, I'm going with that approach.

## Tasks

- [x] Allow creation of projects under categories for category owner/delegate/contributor
- [x] Grant owner rights for project/category for owners of all parent categories
- [x] Disallow giving duplicate role (apart from owner) to inherited owner in subprojects
- [x] Add `get_owners()` to get all owners
- [x] Send email to parent owner if another user creates a project
- [x] Limit owner to current user if not owner of parent (1)
- [x] Add UI hint to the project members page
- [x] Hide duplicate role from UI (do NOT remove yet, breaks remote sync, add task to v0.9)
- [x] Update inherited owners in iRODS via taskflow
- [x] Sync inherited owners during owner transfer in iRODS via taskflow
- [x] ~~Sync inherited owner perms in taskflow when setting project owner via API view~~ (handled with #511)
- [x] Update `synctaskflow` to update inherited roles (adding/removing in taskflow, NOT removing duplicates in the SODAR Core database yet (see #506))
- [x] Update current tests to work
- [x] Also update API view tests with tests for inherited owners
- [x] Add new tests for owner inheritance where necessary
- [x] Document! (also in the projectroles role documentation)

## Notes

- (1) If this isn't done, the user can create a project to which they have no rights to. This is illogical and doesn't look wery nice in the UI either. They can still transfer ownership later if they choose to.
